src/platforms/esp32/components/avm_sys/sys.c

```diff
@@ -167,21 +167,30 @@ void sys_time(struct timespec *t)
     }
 
     t->tv_sec = tv.tv_sec;
-    t->tv_nsec = tv.tv_usec * 1000;
+    t->tv_nsec = (uint_least64_t) tv.tv_usec * 1000;
 }
 
 void sys_monotonic_time(struct timespec *t)
```

Comment:
Just for the sake of discussing further improvements: do we still need this function around since we have `sys_monotonic_time_u64` function?

Reply:
We needed three functions, some of which calling the others depending on platforms.
This PR updates the documentation for each function a little bit.
We now have:
- sys_time for `erlang:system_time/1`
- sys_monotonic_time for `erlang:monotonic_time/1`
- sys_monotonic_time_u64, sys_monotonic_time_ms_to_u64 and sys_monotonic_time_u64_to_ms for timers.

We could arguably replace sys_monotonic_time with sys_monotonic_time_u64 and sys_monotonic_time_u64_to_ms, but this means converting back to timespec. Indeed, some platforms have native timespec, native integers or even native double (emscripten). sys_monotonic_time_u64 should always use the native (fastest) format.